core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java

```diff
@@ -225,7 +225,7 @@ public void testFineGrainedSettings() throws IOException {
             } while (scriptContextSettings.containsKey(scriptContext));
             scriptContextSettings.put(scriptContext, randomFrom(ScriptMode.values()));
         }
-        int numEngineSettings = randomIntBetween(0, 10);
+        int numEngineSettings = randomIntBetween(0, ScriptType.values().length * scriptContexts.length);
```

Comment:
while you are at it can you remove the randomization on script engine service given that we have only one now? seems like this test is a bit outdated after recent changes.